### PR TITLE
Pack continuous votes

### PR DESCRIPTION
**Pull request by PJColombo**
This PR add some changes to the voting contract  `vote()` function to pack both the vote id and the vote percentage into the same parameter to allow backward compatibility.

It also implements unit tests for the continuous vote feature as well as the aforementioned parameter encoding change.

### CODE REVIEW

File: contracts/test/TestImports.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
+pragma solidity 0.4.24;
+
+import "@aragon/os/contracts/acl/ACL.sol";
+import "@aragon/os/contracts/kernel/Kernel.sol";
+import "@aragon/os/contracts/factory/DAOFactory.sol";
+import "@aragon/os/contracts/factory/EVMScriptRegistryFactory.sol";
+
+// You might think this file is a bit odd, but let me explain.
```
Comment (kanaagawa) [inline contracts/test/TestImports.sol:8]: Maybe we can update this comment to not mention truffle?

File: package.json [hunk 2]
```diff
@@ -30,7 +30,6 @@
     "@openzeppelin/contracts": "^3.2.0"
   },
   "devDependencies": {
-    "@1hive/contract-helpers-test": "^0.1.2",
```
Comment (kanaagawa) [inline package.json:33]: Glad we are able to deprecate this utility 🙌

File: test/helpers/evmscript.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const { defaultAbiCoder } = utils;
+
+export const CALLSCRIPT_ID = "0x00000001";
+
+export const encodeCallScript = (
+  actions: { to: string; data: string }[]
+): string => {
+  return actions.reduce((script: string, { to, data }) => {
+    const address = defaultAbiCoder.encode(["address"], [to]);
```
Comment (kanaagawa) [inline test/helpers/evmscript.ts:11]: Probably you already have it in mind. I do not remember if the abi coder returned a byte string 0x prefixed? 🤔 if that the case we may need to trim that.

**Review by 0xGabi — COMMENTED** [review]
Austanding work! 💪 Utilities going to be super useful in the future. I still need to have a in detail review to the tests.

File: test/helpers/evmscript.ts (hunk 3 above)
Comment (PJColombo) [reply test/helpers/evmscript.ts:11]: It does, but they are trimmed a couple of lines [below](https://github.com/BlossomLabs/curve-aragon-voting/blob/4edd2c51d7b3907668b2da536f5520f93847991e/test/helpers/evmscript.ts#L17)  when we performed the `slice()` calls

File: contracts/Voting.sol [hunk 4]
```diff
@@ -279,29 +284,26 @@ contract Voting is IForwarder, AragonApp, BasicMetaTransaction {
...
+    * |    128b       |   64b   |   64b     |
+    * @param _supports Whether voter supports the vote (preserved for backward compatibility purposes)
     * @param _executesIfDecided Whether the vote should execute its action if it becomes decided
     */
-    function vote(uint256 _voteId, bool _supports, bool _executesIfDecided) external voteExists(_voteId) {
-        require(_canVote(_voteId, msgSender()), ERROR_CAN_NOT_VOTE);
-        _vote(_voteId, _supports ? PCT_BASE : 0, msgSender(), _executesIfDecided);
-    }
+    function vote(uint256 _voteData, bool _supports, bool _executesIfDecided) external voteExists(_decodeData(_voteData, 128, MAX_UINT_128)) {
+        uint256 voteId = _decodeData(_voteData, 128, MAX_UINT_128);
```
Comment (michwill) [inline contracts/Voting.sol]: Uhm... wouldn't it be better to keep the voteId in the lower bits? Eg.:
```
voteId = _voteData & MASK_UINT128;
```

Comment (michwill) [reply contracts/Voting.sol]: I would propose this:
```
voteId = _voteData & MASK_UINT128;
support = (_voteData >> (128)) & MASK_UINT64;
quorum = (_voteData >> (128 + 64)) & MASK_UINT64;
if (support == 0 & quorum == 0) {
    // voting as if just voteId was passed...
```

**Pull request merged**